src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -205,72 +206,129 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
+        /// <returns>A dictionary of property names and their values.</returns>        
+        public static Dictionary<string, string> LookupProjectProperties(string projectLocation, params string[] propertyNames)
         {
-            var process = new Process();
-            var output = string.Empty;
-            var msbuildProcessFailed = false;
-            try
+            var projectFile = FindProjectFileInDirectory(projectLocation);
+            var properties = new Dictionary<string, string>();
+            var arguments = new List<string>
             {
-                process.StartInfo = new ProcessStartInfo()
+                "msbuild",
+                projectFile,
+                "-nologo",
+                $"--getProperty:{string.Join(',', propertyNames)}"
+            };
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
                 {
                     FileName = "dotnet",
-                    Arguments = $"msbuild {projectLocation} -getProperty:OutputType",
+                    Arguments = string.Join(" ", arguments),
                     RedirectStandardOutput = true,
+                    RedirectStandardError = true,
                     UseShellExecute = false,
-                    WindowStyle = ProcessWindowStyle.Hidden
-                };
-
+                    CreateNoWindow = true
+                }
+            };
+            try
+            {
                 process.Start();
-                output = process.StandardOutput.ReadToEnd();
-                var hasExited = process.WaitForExit(5000);
+                string output = process.StandardOutput.ReadToEnd();
+                string error = process.StandardError.ReadToEnd();
+                process.WaitForExit(5000);
 
-                // If it hasn't completed in the specified timeout, stop the process and give up
-                if (!hasExited) 
+                if (process.ExitCode == 0)
                 {
-                    process.Kill();
-                    msbuildProcessFailed = true;
-                }
+                    using JsonDocument doc = JsonDocument.Parse(output);
+                    JsonElement root = doc.RootElement;
+                    JsonElement propertiesElement = root.GetProperty("Properties");
 
-                // If it has completed but unsuccessfully, give up
-                if (process.ExitCode != 0)
+                    foreach (var property in propertyNames)
+                    {
+                        if (propertiesElement.TryGetProperty(property, out JsonElement propertyValue))
+                        {
+                            properties[property] = propertyValue.GetString();
+                        }
+                    }
+                }
+                else
                 {
-                    msbuildProcessFailed = true;
+                    Console.WriteLine($"MSBuild process exited with code {process.ExitCode}. Error: {error}");
```

Comment:
No `Console.WriteLine` calls are used. This same code is used inside Visual Studio and there is no console there. If you want to log something you need to pass in an instance if `IToolLogger` to handle logging.

Reply:
I'm not too sure we should be logging anything visible to the user here. We are exposing an implementation detail that would be more confusing if anything.

Reply:
i just removed the logging all together